### PR TITLE
Decompiler Refactor, Introjs addition

### PR DESCRIPTION
**Pull request by Fancy11111**
- refactored decompiler logic
  - added enums
  - split up parsing and interpreting
- decompiler page
  - added introjs
  - added a table that explains singular values of one expression
- searchbar
  - styling
- truth table
  - added introjs
- intro component
  - created intro.ts, capsules introjs calls

### CODE REVIEW

**Comment by Fancy11111** [comment]
version bump before merging probably

File: src/assets/decompiler.ts [hunk 1]
```diff
@@ -2,16 +2,59 @@ export interface ParsedInstruction {
...
+export enum Shift {
+  Never = 0,
+  L = 1,
+  R = 2,
+  Invalid = 3
+}
+
+export enum ReadWrite {
+  Read = 1,
+  Write = 0
```
Comment (stefnotch) [inline src/assets/decompiler.ts:57]: Damn sweet, that's one well chosen use of Typescript enums!

File: src/components/CollapsibleTable.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<template>
+  <!-- Or I could just have used a datalist -->
+  <details open>
+    <summary>
+      <slot name="title"></slot>
```
Comment (stefnotch) [inline src/components/CollapsibleTable.vue:5]: Oh nice, slots are that easy to use? Looks like I'm learning something new 👍

File: vite.config.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default {
   plugins: [
     vue(),
     copy({
-      assets: ["node_modules/monaco-editor/min/vs"],
+      assets: ["node_modules/monaco-editor/min/vs", "node_modules/intro.js/minified"],
```
Comment (stefnotch) [inline vite.config.js:13]: I can apparently get away without that. I'll try getting rid of it, please report back if my changes work or if they horribly break everything.

**Review by stefnotch — APPROVED** [review]

File: src/assets/decompiler.ts (hunk 1 above)
Comment (Fancy11111) [reply src/assets/decompiler.ts:57]: the read write one maybe a bit overkill, but the other ones definitely feel a lot better
ReadWrite is not really a toggle so it made more sense as a enum to me

File: src/components/Intro.vue [hunk 4]
```diff
@@ -4,13 +4,14 @@
...
 import { defineComponent, ref} from "vue";
+// @ts-ignore (shut up, dear compiler-chan)
 import introJS from 'intro.js';
 
 export default defineComponent({
   props: {
     text: {
       type: String,
-      requried: false,
+      required: false,
```
Comment (Fancy11111) [inline src/components/Intro.vue:14]: funny that typescript didn't cry about that one

File: src/components/CollapsibleTable.vue (hunk 2 above)
Comment (Fancy11111) [reply src/components/CollapsibleTable.vue:5]: https://vuejs.org/v2/guide/components-slots.html#Named-Slots

File: vite.config.js (hunk 3 above)
Comment (Fancy11111) [reply vite.config.js]: works fine for me, I wasn't quite sure at first tho, so I did that just to be sure

**Pull request merged**